src/Exceptions/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+set(BOT_LIB_NAME Exceptions)
+file(GLOB_RECURSE LibSource ./*.cpp)
```

Comment:
`LibSource` doesn't seem to be used(?). The recommended way is to not use globbing, but instead add all source files explicitly (as you have done here).